src/appleseed/renderer/modeling/aov/pixeltimeaov.cpp

```diff
@@ -181,38 +184,12 @@ namespace
         void post_process_image(const Frame& frame) override
         {
             const AABB2u& crop_window = frame.get_crop_window();
+            ColorMap color_map;
 
-            // Find the maximum value.
+            float min_time = numeric_limits<float>::max();
             float max_time = 0.0f;
-
-            for (size_t j = crop_window.min.y; j <= crop_window.max.y; ++j)
-            {
-                for (size_t i = crop_window.min.x; i <= crop_window.max.x; ++i)
-                {
-                    float val;
-                    m_image->get_pixel(i, j, &val);
-                    max_time = max(val, max_time);
-                }
-            }
-
-            if (max_time == 0.0f)
-                return;
-
-            const float rcp_max_time = 1.0f / max_time;
-
-            // Normalize.
-            for (size_t j = crop_window.min.y; j <= crop_window.max.y; ++j)
-            {
-                for (size_t i = crop_window.min.x; i <= crop_window.max.x; ++i)
-                {
-                    float c;
-                    m_image->get_pixel(i, j, &c);
-
-                    c *= rcp_max_time;
-
-                    m_image->set_pixel(i, j, &c);
-                }
-            }
+            color_map.find_min_max_red_channel(*m_image, crop_window, min_time, max_time);
```

Comment:
I'm sorry but I've raised this point at twice already and you seem to ignore it:

**You are not loading a palette into the color map here.** I just don't understand how it can work. Am I missing something? (It can definitely happen.)

Can you please post renders for all the AOVS you modified, in particular Pixel Time?

Reply:
There was a slight confusion, addressed on Discord.